pkg/apiservers/kubeapiserver.go

```diff
@@ -0,0 +1,220 @@
+package apiservers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorV1 "github.com/openshift/api/operator/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// KubeAPIServerBuilder provides struct for kubeAPIServer object.
+type KubeAPIServerBuilder struct {
+	// KubeApiServer definition. Used to create an kubeAPIServer object.
+	Definition *operatorV1.KubeAPIServer
+	// Created kubeAPIServer object.
+	Object *operatorV1.KubeAPIServer
+	// apiClient opens api connection to the cluster.
+	apiClient goclient.Client
+	// Used in functions that define or mutate kubeAPIServer definition. errorMsg is processed before the
+	// kubeAPIServer object is created.
+	errorMsg string
+}
+
+var kubeAPIServerObjName = "cluster"
+
+// Exists checks whether the given kubeAPIServer exists.
+func (builder *KubeAPIServerBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to collect kubeAPIServer object due to %s", err.Error())
+	}
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// PullKubeAPIServerBuilder pulls existing kubeApiServer from the cluster.
+func PullKubeAPIServerBuilder(apiClient *clients.Settings) (*KubeAPIServerBuilder, error) {
+	builder := KubeAPIServerBuilder{
```

Comment:
Can you add a check here to ensure apiClient isn't nil? (ref: https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/lca/imagebasedupgrade.go#L75)

Reply:
done